File: Building.java
/**
 * Abstract class which every building extends from
 */
public abstract class Building extends Quacks {
    private final int BASE_COST;
    private final double MULTIPLICITY = 0.15; // value that cost of building increases by (ex. 15% or 0.15)
    private final double BASE_QPS;
    private double qps; // quacks per second
    private double qpc; // quacks per click
    
    public Building(int cost, double qps) {
        super();
        BASE_COST = cost;
        BASE_QPS = qps;
    }
    
    /**
     * Takes building count and raises 1.15 to the power of building count. Returns the result
     * 
     */
    public double multiplier(int building_count) {
        double multiplier = Math.pow(1.15, building_count);
        return multiplier;
    }
}

File: Building.js
/**
 * Abstract class which every building extends from
 */
class Building extends Quacks {
  constructor(cost, qps) {
    super(); // calls Quacks constructor
    if (new.target === Building) {
      throw new Error("abstract class!! can't instantiate directly");
    }

    this.BASE_COST = cost; // base cost of the building
    this.MULTIPLICITY = 0.15; // cost increase rate (15%)
    this.BASE_QPS = qps; // base quacks per second
    this.qps = qps; // current quacks per second
    this.qpc = 0; // quacks per click (default 0)
    this.amount = 0; // number of buildings owned
  }

  /**
   * Takes building count and raises 1.15 to the power of building count. Returns the result.
   */
  multiplier(buildingCount) {
    return Math.pow(1.15, buildingCount);
  }

  getAmt() {
    return this.amount;
  }

  addAmt(amt = 1) {
    this.amount += amt;
  }

  getQPS() {
    return this.qps;
  }

  getQPC() {
    return this.qpc;
  }
}
